cgi.go

```diff
@@ -161,23 +132,101 @@ func computeKnownVariables(request *http.Request, p *runtime.Pinner) (cArr [27]C
 		}
 	}
 
-	setKnownServerVariable(p, &cArr, serverName, reqHost)
+	registerTrustedVar(knownServerKeys["SERVER_NAME\x00"], reqHost, track_vars_array, thread)
 	if reqPort != "" {
-		setKnownServerVariable(p, &cArr, serverPort, reqPort)
+		registerTrustedVar(knownServerKeys["SERVER_PORT\x00"], reqPort, track_vars_array, thread)
+	} else {
+		registerTrustedVar(knownServerKeys["SERVER_PORT\x00"], "", track_vars_array, thread)
 	}
 
 	// Variables defined in CGI 1.1 spec
 	// Some variables are unused but cleared explicitly to prevent
 	// the parent environment from interfering.
 
-	// These values can not be override
-	setKnownServerVariable(p, &cArr, contentLength, request.Header.Get("Content-Length"))
-	setKnownServerVariable(p, &cArr, gatewayInterface, "CGI/1.1")
-	setKnownServerVariable(p, &cArr, serverProtocol, request.Proto)
-	setKnownServerVariable(p, &cArr, serverSoftware, "FrankenPHP")
-	setKnownServerVariable(p, &cArr, httpHost, request.Host) // added here, since not always part of headers
+	// These values can not be overridden
+	registerTrustedVar(knownServerKeys["CONTENT_LENGTH\x00"], request.Header.Get("Content-Length"), track_vars_array, thread)
```

Comment:
Maybe should we validate the date (it must be a positive int) here instead of escaping it or passing it blindly. It looks like there are issues with FPM support in Caddy because of this lack of validation, we may have the same issue: https://github.com/caddyserver/caddy/pull/6639/files#diff-704d0cf138fc604375b83fd6eab74382b1277998511a0cc2e2876644346fa2d5R229-R232

Reply:
I tested this case in `caddy_test.go`. It looks like Caddy already ignores the user-defined content-length header.

Reply:
Caddy DOES validate that the Content-Length is an integer, otherwise the validation of the Content-Length header [here](https://github.com/dunglas/frankenphp/blob/17e57287eb10b4431db58c3b612dad7ca1075f07/frankenphp.go#L421) would straight up crash the server [here](https://github.com/dunglas/frankenphp/blob/17e57287eb10b4431db58c3b612dad7ca1075f07/frankenphp.go#L601). Probably not a good place to panic in either case